doc/sendrequest.md

```diff
@@ -54,6 +54,33 @@ will reflect cancellation only after the request's transmission is avoided or th
 cancellation result. Otherwise, the `Task` will still complete or fault based on the server's ordinary
 response to the original request.
 
+## Responses come back asynchronously
+
+As JSON-RPC is fundamentally an async protocol, and typically used to communicate between processes or machines,
+`JsonRpc` exposes only async methods for sending requests and awaiting responses.
+
+In fact, supporting non-Task returning methods would be potentially very problematic because that would mean StreamJsonRpc is responsible to artificially block the calling thread until the result came back. But that can quickly lead to hangs if, for example, the calling thread is your app's UI thread. It can deadlock if the server needs to call back to your client before returning a response, and your client can't respond because its UI thread is hung.
+
+When you have an unavoidable requirement to synchronously block while waiting for *any* async code to complete (whether JSON-RPC related or otherwise),
+we recommend use of the [JoinableTaskFactory](https://aka.ms/vsthreading) as found in the [Microsoft.VisualStudio.Threading](https://www.nuget.org/packages/Microsoft.VisualStudio.threading) library. That can block the UI thread (or any other thread) while executing async operations. In the case of StreamJsonRpc, that means the client might have something like this:
+
+```cs
+interface IJsonRpcService
+{
+    Task SomeOperationAsync();
+}
+
+var client = JsonRpc.Attach<IJsonRpcService>(pipe);
```

Comment:
Is there any requirement for JsonRPC to be JTF aware, for example in the scenario you provided above? I am guessing no but wanted to check.

When server calls back the client via JsonRPC, would their callback always execute at thread pool so it would still be client's callback handler responsibility to use JTF to switch to main thread if needed?

Reply:
StreamJsonRpc already references the threading library, but it isn't JTF aware. It's just typical async code that a JTF aware client could call into and block. But if there were callbacks that needed to access a UI thread blocked by JTF.Run, they'd need to do all the necessary joining to avoid the deadlock.

Reply:
And yes, StreamJsonRpc uses the threadpool to listen for incoming messages, and invocations based on those messages are made via a `SynchronizationContext` that the owner provides (or we use the default one, which schedules to the thread pool).